src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -260,26 +269,48 @@ function show() {
   }
 }
 
+function waitForPopoverFocus() {
+  return new Promise((resolve) => {
+    requestAnimationFrame(() => {
+      requestAnimationFrame(() => {
+        requestAnimationFrame(resolve);
```

Comment:
I see this code not yours, but why we use here this chain of RAF's?

Reply:
because before using all the callbacks after the opening of the dropdown we need to wait for it to be opened. so this is the solution it was found

Reply:
Why don't use `nextTick` here?

Reply:
I'm not sure where the comment documenting this hack went, I remember writing one.

Every time we open the ec-popover (which uses v-popover from floating-vue), they try to get a focus on the floating div after it's open. But they don't do it immediately, but after 2 RAFs (they have an util for it: https://github.com/Akryum/floating-vue/blob/main/packages/floating-vue/src/util/frame.ts#L1). This breaks lots of our functionality, because we want to focus on first item in the ec-popover like search or CTA if they are available. If we just focus in afterShow, the focus will be moved to div 2 RAFs after. So the trick is to wait 3 frames and bypass their nonsense. That's why this function does what it does.